food-suggestor/src/utils/ApiUtility.jsx

```diff
@@ -0,0 +1,31 @@
+import axios from 'axios'
+
+//     /*
+//      * Lists 
+//      * https://www.themealdb.com/api/json/v1/1/list.php?c=list- list all categories
+//      * https://www.themealdb.com/api/json/v1/1/list.php?a=list - list all areas
+//      * https://www.themealdb.com/api/json/v1/1/list.php?i=list - list all ingredients
+//      * 
+//      * Filters
+//      * https://www.themealdb.com/api/json/v1/1/filter.php?c=input - filter by category
+//      * https://www.themealdb.com/api/json/v1/1/filter.php?a=input - filter by area
+//      * https://www.themealdb.com/api/json/v1/1/filter.php?i=input - filter by main ingredient
+//      */
+
+const url = "https://www.themealdb.com/api/json/v1/1/";
+
+const getData = (resource, filterFlag, input) => {
```

Comment:
It is cleaner and more maintainable to create two functions for accessing the two distinct API endpoints.

Currently `getData()` is responsible for any API call, and the developer has to manage `filterFlag` and `input` which add complexity since the only real necessary variation occurs in choosing the correct `resource`.

Reply:
thanks! pushed new changes